src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -116,34 +118,63 @@ public void SslStream_StreamToStream_Alpn_Success(List<SslApplicationProtocol> c
-                    ServerCertificate = certificate,
-                };
-                
-                Task t1 = Assert.ThrowsAsync<AuthenticationException>(() => client.AuthenticateAsClientAsync(clientOptions, CancellationToken.None));
-                Task t2 = Assert.ThrowsAsync<AuthenticationException>(() => server.AuthenticateAsServerAsync(serverOptions, CancellationToken.None));
-
-                Assert.True(Task.WaitAll(new[] { t1, t2 }, TestConfiguration.PassingTestTimeoutMilliseconds));
+                    Task<TcpClient> serverTask = listener.AcceptTcpClientAsync();
+                    await client.ConnectAsync(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndpoint).Port);
+
+                    using (TcpClient server = await serverTask)
+                    using (SslStream serverStream = new SslStream(server.GetStream(), leaveInnerStreamOpen: false))
+                    using (SslStream clientStream = new SslStream(client.GetStream(), leaveInnerStreamOpen: false))
+                    using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+                    {
+                        SslServerAuthenticationOptions serverOptions = new SslServerAuthenticationOptions
+                        {
+                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http2 },
+                            ServerCertificate = certificate,
+                        };
+                        SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions
+                        {
+                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http11 },
+                            RemoteCertificateValidationCallback = AllowAnyServerCertificate,
+                            TargetHost = certificate.GetNameInfo(X509NameType.SimpleName, false),
+                        };
+
+                        // Test alpn failure only on platforms that supports ALPN.
+                        if ((RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && !(PlatformDetection.IsUbuntu1404 || PlatformDetection.IsDebian8 || PlatformDetection.IsCentos6)) ||
+                            (PlatformDetection.IsWindows && !PlatformDetection.IsWindows7))
+                        {
+                            Task t1 = Assert.ThrowsAsync<IOException>(() => clientStream.AuthenticateAsClientAsync(clientOptions, CancellationToken.None));
+                            Task t2 = Assert.ThrowsAsync<AuthenticationException>(() => serverStream.AuthenticateAsServerAsync(serverOptions, CancellationToken.None)).ContinueWith(t =>
+                            {
+                                server.Dispose();
+                            }, TaskScheduler.Default);
+
+                            Assert.True(Task.WaitAll(new[] { t1, t2 }, TestConfiguration.PassingTestTimeoutMilliseconds));
```

Comment:
nit Task.WaitAll

you can now use 

` await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1, t2);`